modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java

```diff
@@ -1789,21 +1789,18 @@ public CodegenModel fromModel(String name, Schema schema) {
             List<String> allRequired = new ArrayList<String>();
 
             // if schema has properties outside of allOf/oneOf/anyOf also add them to m
-            if (schema.getProperties() != null) {
-                addVars(m, unaliasPropertySchema(schema.getProperties()), schema.getRequired(), null, null);
+            if (composed.getProperties() != null && !composed.getProperties().isEmpty()) {
+                if (composed.getOneOf() != null && !composed.getOneOf().isEmpty()) {
+                    LOGGER.warn("'oneOf' is intended to include only the additional optional OAS extension discriminator object. " +
+                            "For more details, see https://json-schema.org/draft/2019-09/json-schema-core.html#rfc.section.9.2.1.3 and the OAS section on 'Composition and Inheritance'.");
+                }
+                addVars(m, unaliasPropertySchema(composed.getProperties()), composed.getRequired(), null, null);
             }
 
-            // uncomment this when https://github.com/swagger-api/swagger-parser/issues/1252 is resolved
-            // if schema has additionalproperties outside of allOf/oneOf/anyOf also add it to m
-            // if (schema.getAdditionalProperties() != null) {
-            //     addAdditionPropertiesToCodeGenModel(m, schema);
-            // }
-
             // parent model
             final String parentName = ModelUtils.getParentName(composed, allDefinitions);
             final List<String> allParents = ModelUtils.getAllParentsName(composed, allDefinitions);
             final Schema parent = StringUtils.isBlank(parentName) || allDefinitions == null ? null : allDefinitions.get(parentName);
-            final boolean hasParent = StringUtils.isNotBlank(parentName);
```

Comment:
Are we deleting this because it is unused?

Reply:
Yeah. This would fail compilation of it was still expected.